pkg/cloudprovider/cloudprovider.go

```diff
@@ -291,14 +287,14 @@ func (c *CloudProvider) resolveInstanceTypes(ctx context.Context, nodeClaim *cor
 	}
 	reqs := scheduling.NewNodeSelectorRequirements(nodeClaim.Spec.Requirements...)
 	return lo.Filter(instanceTypes, func(i *cloudprovider.InstanceType, _ int) bool {
-		return reqs.Compatible(i.Requirements) == nil &&
+		return reqs.Compatible(i.Requirements, lo.Ternary(nodeClaim.IsMachine, scheduling.AllowUndefinedWellKnownLabelsV1Alpha5, scheduling.AllowUndefinedWellKnownLabelsV1Beta1)) == nil &&
```

Comment:
Can you remind me why we're adding more requirements checks here?

Reply:
It's not more compatibility checks. It's a change in the compatibility logic which assumes that keys that are undefined are denied by default, unless you specify an option that tells us to accept certain undefined keys